packages/ketcher-react/src/script/ui/views/components/ContextMenu/ContextMenuTrigger.tsx

```diff
@@ -0,0 +1,111 @@
+/****************************************************************************
+ * Copyright 2022 EPAM Systems
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ***************************************************************************/
+
+import { FunctionalGroup } from 'ketcher-core'
+import { useCallback } from 'react'
+import { useContextMenu } from 'react-contexify'
+import { useAppContext } from 'src/hooks'
+import Editor from 'src/script/editor'
+import { ContextMenuItemProps, CONTEXT_MENU_ID } from './ContextMenu'
+
+const ContextMenuTrigger: React.FC = ({ children }) => {
+  const { getKetcherInstance } = useAppContext()
+  const { show, hideAll } = useContextMenu<ContextMenuItemProps>({
+    id: CONTEXT_MENU_ID
+  })
+
+  const handleDisplay = useCallback(
+    (event) => {
+      const editor = getKetcherInstance().editor as Editor
+      const ci = editor.findItem(event, ['bonds'])
+
+      if (!ci) {
+        hideAll()
+        return
+      }
+
+      // Resolve conflict with existing functional group context menu
```

Comment:
Could you, please, describe in a few words what a conflict occurs?
Showing two menus at the same time?
Then I propose to move this part of code at separate function.

Reply:
Oh sorry, I described it in the PR description but I didn't put it in the code. I'll put the PR link into the code and move this part to a function as you suggested. Thank you for pointing it out.

Reply:
Here are the code comments: https://github.com/epam/ketcher/pull/1896/files#diff-65d4d0aa8707ab4ab7d93b7f42d0b60f211dbcfd6e5d7b06398fd43a8b1b68f3R31-R48

And I added a GIF to the PR description to show more details.